doc/loganalyzer_to_pytest.md

```diff
@@ -0,0 +1,108 @@
+#### Approach 1
+
+#### Requirements
+Move existed loganalyzer to be supported by pytest framework.
+
+#### Overview
+Current loganalyzer implementation consists of the following parts:
+Init - copy loganalyzer.py to the DUT. Add start marker to the syslog.
+
+Analyze - copy loganalyzer.py and all match,expect,ignore files to the DUT. Extract logs from syslog. Add end marker. Analyze logs based on regexp described in match,expect,ignore files.
+Generate summary and result files.
+
+End - Check if "TOTAL MATCHES != 0" then generate system dump. Store system dump, result and summary files to the ansible host.
+
+In proposed approach script copies loganalyzer.py and files with regexp to the DUT and runs syslog analyze on the DUT.
+
+Current loganalyzer implementation description:
+https://github.com/Azure/SONiC/wiki/LogAnalyzer
+
+**Development**
+**Implement:**
+	Module "loganalyzer.py" with class "Loganalyzer".
+	"Loganalyzer" class - executes shell commands on DUT.
+	**"Loganalyzer" class interface:**
+- __init__(match: list, expect: list, ignore: list)
+- exec_cmd(cmd) - execute shell command on the DUT
+- extract_syslog() - calls "extract_syslog" from "log_processor.py"
+- upload_search_files(dest) - upload match, expect, ignore files to the DUT
+- parse_summary_file() - convert generated on DUT "result.loganalysis.{TEST_NAME}..." file to json.
+- parse_result_file() - convert generated on DUT "summary.loganalysis.{TEST_NAME}" file to json.

```

Comment:
For the methods that are internally used by Loganalyzer, we can consider hide them using the "_" prefix.